src/functions/assertions/Should.ps1

```diff
@@ -122,18 +122,23 @@ function Should {
             $shouldThrow = 'Stop' -eq $PSBoundParameters["ErrorAction"]
         }
 
+        # first check if we are in the context of Pester, if not we will always throw, and won't disable Should:
+        # This check is slightly hacky, here we are reaching out the the caller session state and
+        # look for $______parameters which we know we are using inside of the Pester runtime to
+        # keep the current invocation context, when we find it, we are able to add non-terminating
+        # errors without throwing and terminating the test.
+        $pesterRuntimeInvocationContext = $PSCmdlet.SessionState.PSVariable.GetValue('______parameters')
+        $isInsidePesterRuntime = $null -ne $pesterRuntimeInvocationContext
+        $pesterRuntimeInvocationContext.Configuration.Should.ErrorAction.Value
+
+        if ($isInsidePesterRuntime -and $pesterRuntimeInvocationContext.Configuration.Should.DisableV5.Value) {
```

Comment:
Need to ignore `Invoke`, `InvokeVerifiable` and `HaveParameter` for now. Or delay merge for now.

Reply:
Why? What am I overlooking?

Reply:
My logic was that they are all registered via `Should` function, so failing inside of it should fail for all registered assertions operators.

Reply:
Won't this also throw on e.g. `Should -Invoke`? We don't have a replacement atm.